### PR TITLE
Enable the task unit test projects in helix and apply fixes to them.

### PR DESCRIPTION
**Pull request by marcpopMSFT**
To enable this required a variety of changes:
- Undo the lock file changes that were done for CG. Since it's test only, we can exclude those requirements
- upgrade tests to the current tfm
- pass through the required runtime pack flag to the ProcessFrameworkReference task
- Update the mock to correctly pull existing objects
- Update a bunch of strings to match our naming and comment requirements
- Add error code exclusions to fix those tests
- Disable four tests on non-windows as they weren't working there
- Add environment variable for the assets path and include them in the helix package
- Set environment variables for two other items required for the unit tests to work correctly and remove those from the CLI arguments

### CODE REVIEW

**Comment by marcpopMSFT** [comment]
@sbomer I'm trying to enable these unit tests we haven't had running in a while and they include a test that requires certain naming conventions for various messages. in particular, message either need an error number or need to end in _Info.

I tried to update ILLinkOptimizedAssemblies and ILLinkRunning but it turns out those are consumed by the ILLink.targets that aren't in this repo.

What is the path forward here? Do I need to limit this change to main and make an update in the dotnet/linker repo first? Are there tests in the linker repo that will fail if I do? I also noticed that there's a copy of the linker targets in the runtime repo. Does that also need to be modified or does that get updated automatically through codeflow?

**Comment by marcpopMSFT** [comment]
Thanks for the summary. I think for simplicity, I may just enable the unit test project in main and only fix the strings in runtime (instead of trying to change things in 7.0 as well). If I make a change in runtime to change the ID, will it break tests in runtime that rely on the string resources that live here?

I'm not ready yet to enable the unit test project (still working through infrastructure issues) and I could potentially disable this one test for now (and revert the string changes in this PR) but it'd be good to figure out how we would update that in the future. Would it be something where we'd have to have both the old and new string id, update the SDK version in runtime, then change runtime, then remove the old strings here (ie do a staged change)?

**Comment by marcpopMSFT** [comment]
I talked to Daniel and we decided it probably wasn't worth a bunch of effort to just unwind the two linker resources for a test (as there's no functional justification for that afaik). Instead, I'll add an exception to the test for now and revert those. We can decide to change the linker ones in the future if we decide it's a problem.

**Comment by marcpopMSFT** [comment]
@dsplaisted, finally got it to pass checks. The most interesting of my changes are the inclusion of assets in the helix payload, setting a flag for that and the disabling of some tests on non-windows, and my reverting of the lock files (which may end up triggering CG which can be dismissed). I think the rest of the changes are pretty straightforward. Happy to walk you through it.

**Review by dsplaisted — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenThatWeHaveErrorCodes.cs [hunk 1]
```diff
@@ -32,12 +32,28 @@ public class GivenThatWeHaveErrorCodes
...
             1180,
+            1195,
+            1197,
+            1199,
+            1200,
+            1201,
+            1203,
+            1204,
+            1206,
+            1207
```
[the remark this answers is not in the document]
Comment (marcpopMSFT) [reply src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenThatWeHaveErrorCodes.cs:48]: They are new codes that were added to main. We had an even newer code added in 4xx to not overlap hence the gap.

File: src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenAResolvePackageDependenciesTask.cs [hunk 2]
```diff
@@ -40,11 +40,11 @@ public static IEnumerable<object[]> ItemCounts
                 {
                     new object[] {
                         "dotnet.new",
-                        new int[] { 110, 2536, 1, 845, 75 },
+                        new int[] { 110, 2536, 1, 846, 73 },
```
[the remark this answers is not in the document]
Comment (marcpopMSFT) [reply src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenAResolvePackageDependenciesTask.cs:43]: I do not. want to review the test results together?  There are so many packages listed that it's hard to tell what is new versus what it used to be.

[the remark this answers is not in the document]
Comment (marcpopMSFT) [reply src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenAResolvePackageDependenciesTask.cs:43]: chatted offline and agreed it wasn't worth investigating further.

**Pull request merged**